Month3/LinkedList/LinkedList/TDLList.h

```diff
@@ -0,0 +1,404 @@
+		{
+			return;
+		}
+		if (Index < (Size / 2))
+		{
+
+			for (int CurrentIndex = 0; CurrentIndex = Index; ++CurrentIndex)
+			{
+				Current = Current->Next;
+			}
+			return Current->Next;
+		}
+		else if (Index > (Size / 2))
+		{
+			for (int CurrentIndex = Size - 1; CurrentIndex = Index; --CurrentIndex)
+			{
+				Current = Current->Prev;
+			}
+			return Current->Prev;
+		}
+	}
+
+
+	T& GetHead()
+	{
+		return Head->Element;
+	}
+	const T& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	T& GetTail()
+	{
+		return Tail->Element;
+	}
+	const T& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	const int GetSize() const
+	{
+		return Size;
+	}
+
+	void AddHead(const T& NewElement)
+	{
+		if (Size == 0)
+		{
+			FNode* NewNode = new FNode;
+			NewNode->Element = NewElement;
+
+			Head = NewNode;
+			Tail = NewNode;
+
+			++Size;
+		}
+		else
+		{
+			FNode* NewNode = new FNode;
+			NewNode->Element = NewElement;
+
+			Head->Prev = NewNode;
+			NewNode->Next = Head;
+			Head = NewNode;
+			++Size;
+		}
+	}
+
+	void AddTail(const T& NewElement)
+	{
+		if (Tail) {
+
+			/*FNode* NewNode = new FNode;
+			NewNode->Element = NewElement;
+			NewNode->Next = nullptr;
+			Tail->Next = NewNode;
+			NewNode->Prev = Tail;
+			Tail = NewNode;
+			Size++;*/
+
+			Tail->Next = new FNode;
+			Tail->Next->Element = NewElement;
+			Tail->Next->Prev = Tail;
+			Tail = Tail->Next;
+			Size++;
+		}
+		else
+		{
+			AddHead(NewElement);
+		}
+	}
+
+
+
+
+	void Insert(const T& NewElement, const int Index)
+	{
+		if (Index < 0 || Index > Size)
+		{
+			return;
+		}
+
+		else if (Index == 0)
+		{
+			AddHead(NewElement);
+		}
+
+		else if (Index == Size)
+		{
+			AddTail(NewElement);
+		}
+
+		else
+		{
+			FNode* NewNode = new FNode;
+			NewNode->Element = NewElement;
+			FNode* Current = Head;
+
+			for (int Currentindex = 0; Currentindex < Index; ++Currentindex)
+			{
+				Current = Current->Next;
+			}
+			NewNode->Next = Current;
+			NewNode->Prev = Current->Prev;
+			Current->Prev->Next = NewNode;
+			Current->Prev = NewNode;
+			Size++;
+		}
+
+
+
+	}
+
+
+
+	void RemoveTail()
+	{
+		FNode* Current = Tail;
+		Tail = Tail->Prev;
+		delete Current;
+		--Size;
+	}
+
+	void Remove(const int Index)
+	{
+		if (Index >= Size || Index < 0 || Size == 0)
+		{
+			return;
+		}
+
+		else if (Size == 1)
+		{
+			Clear();
+		}
+		else if (Index == (Size - 1))
+		{
+			RemoveTail();
+
+		}
+		else
+		{
+			FNode* Current = Head;
+
+			for (int CurrentIndex = 0; CurrentIndex < Index - 1; CurrentIndex++)
+			{
+				Current = Current->Next;
+			}
+
+			FNode* After = Current->Next;
+			Current->Next = After->Next;
+			After->Next = nullptr;
+			After->Prev = nullptr;
+			delete After;
+			Current->Next->Prev = Current;
+			Size--;
+		}
+
+	}
+
+
+
+	void Clear()
+	{
+		for (int CurrentIndex = 0; CurrentIndex < Size; ++CurrentIndex)
+		{
+			FNode* Current = Head;
+			Head = Head->Next;
+			delete Current;
```

Comment:
Hmm you're not setting the new Head previous element to nullptr, but i guess it doesn't hurt anybody since they're all gonna get deleted straight away